includes/class-sensei-course.php

```diff
@@ -3313,6 +3316,44 @@ public static function prerequisite_complete_message() {
 		}
 	}
 
+	/**
+	 * Log an event when a course is initially published, but not on
+	 * subsequently publishes (i.e. if the course is unpublished and
+	 * re-published, do not log another event).
+	 *
+	 * @since 2.1.0
+	 * @access private
+	 *
+	 * @param string  $new_status The new post status.
+	 * @param string  $old_status The old post status.
+	 * @param WP_Post $post       The course.
+	 */
+	public function log_initial_publish_event( $new_status, $old_status, $post ) {
+		if ( 'course' !== $post->post_type ) {
```

Comment:
You should add a check for `|| $new_status === $old_status` here. This hook is called even for `publish -> publish`